src/Microsoft.ServiceFabric.Client.Http/LocalImageStoreClient.cs

```diff
@@ -0,0 +1,274 @@
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public Task UploadFileAsync(
+            byte[] fileContentsToUpload,
+            string pathInImageStore,
+            long? serverTimeout = 60,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.UploadFileAsync(
+                fileContentsToUpload,
+                pathInImageStore,
+                null,
+                serverTimeout,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Uploads contents of the file to the image store.
+        /// Used by IApplicationClient.UploadApplicationPackageAsync to trace a request Id associated with original Uplaod call.
+        /// </summary>
+        /// <param name="fileContentsToUpload">File contents to upload.</param>
+        /// <param name ="pathInImageStore">Relative path to file or folder in the image store from its root.</param>
+        /// <param name="requestId">Request Id to use for tracing.</param>
+        /// <param name ="serverTimeout">The server timeout for performing the operation in seconds. This specifies the time
+        /// duration that the client is willing to wait for the requested operation to complete. The default value for this
+        /// parameter is 60 seconds.</param>
+        /// <param name ="cancellationToken">Cancels the client-side operation.</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// </returns>
+        internal Task UploadFileAsync(
+            byte[] fileContentsToUpload,
+            string pathInImageStore,
+            string requestId = null,
+            long? serverTimeout = 60,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+
+            pathInImageStore.ThrowIfNull(nameof(fileContentsToUpload));
+            pathInImageStore.ThrowIfNull(nameof(pathInImageStore));
+            File.WriteAllBytes(Path.Combine(imageStorePath, pathInImageStore), fileContentsToUpload);
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public Task UploadFileChunkAsync(
+            byte[] fileChunkToUpload,
+            string pathInImageStore,
+            Guid? sessionId,
+            long startBytePosition,
+            long endBytePosition,
+            long length,
+            long? serverTimeout = 60,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.UploadFileChunkAsync(
+                fileChunkToUpload,
+                pathInImageStore,
+                sessionId,
+                endBytePosition,
+                endBytePosition,
+                length,
+                null,
+                serverTimeout,
+                cancellationToken);
+        }
+
+        internal  Task UploadFileChunkAsync(
```

Comment:
As mentioned above not all methods in IIMageStoreClient apply here, Upload file chunk doesnt applyt o FileStore as there is no one on the share to assemblt these files

Reply:
i realize that but i have to implemented them somehow since they are part of the interface, or do you mean we should throw a NotSupportedException instead?

Reply:
Please see above, 2 different implementations are not desired here. In this case if statement in Upload and delete methods would be better. ImageStoreClient doesn't need to determine the connection string unless a method to Upload and Delete is called on it, then it can just cache that value.